app/grandchallenge/components/models.py

```diff
@@ -2397,25 +2421,38 @@ def create_civ_for_file(
     def get_civ_for_interface(self, interface):
         raise NotImplementedError
 
-    def get_current_value_for_interface(
-        self, *, interface, user_upload=None, upload_session=None
-    ):
+    def get_current_value_for_interface(self, *, interface, user):
         try:
             return self.get_civ_for_interface(interface=interface)
         except ObjectDoesNotExist:
             return None
         except MultipleObjectsReturned as e:
-            if user_upload:
-                user_upload.handle_file_validation_failure(
-                    error_message="An unexpected error occurred"
-                )
-            elif upload_session:
-                upload_session.update_status(
-                    error_message="An unexpected error occurred",
-                    status=RawImageUploadSession.FAILURE,
-                )
+            error_handler = self.get_error_handler()
+            error_handler.handle_error(
+                interface=interface,
+                error_message="An unexpected error occurred",
+                user=user,
+            )
             raise e
 
+    def get_error_handler(self, *, linked_object=None):
+        # local imports to prevent circular dependency
+        from grandchallenge.algorithms.models import Job
+
+        if linked_object and isinstance(linked_object, RawImageUploadSession):
+            return RawImageUploadSessionCIVErrorHandler(
+                upload_session=linked_object,
+                linked_job=self if isinstance(self, Job) else None,
+            )
+        elif isinstance(self, Job):
+            return JobCIVErrorHandler(job=self)
+        elif linked_object and isinstance(linked_object, UserUpload):
+            return UserUploadCIVErrorHandler(
+                user_upload=linked_object,
+            )
+        else:
+            return SystemCIVErrorHandler()
```

Comment:
We don't have a fallthrough for new/future situations that the developer may need to implement their own specific handler for. Can we already determine the cases where `SystemCIVErrorHandler` is appropriate and raise an error in else?